### PR TITLE
feat(web): atlas-integration

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
## PR-Codex overview
This PR focuses on refactoring and enhancing the codebase by removing unused files, updating environment variables, adding a new utility function, and improving component functionality with better state management and user feedback.

### Detailed summary
- Deleted several unused files in `web/src` and `web/netlify/functions`.
- Updated environment variables in `.env.testnet.public` and `.env.local.public`.
- Added `isEmpty` utility function to check for empty strings.
- Refactored `EnsureAuth` component to use `useAtlasProvider`.
- Enhanced `SubmitItem` and `SubmitList` components to utilize the new `isEmpty` function.
- Improved file upload handling in `ImageInput`, `FileInput`, and `LogoUpload` components.
- Added `EmailVerificationInfo` component for email verification status.
- Updated `Settings` page to include routing for email confirmation.
- Enhanced user feedback with success and error toasts on actions.

> The following files were skipped due to too many changes: `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Release Notes

- **New Features**
	- Introduced the `AtlasProvider` for enhanced context management.
	- Added a new `EmailConfirmation` component for email verification processes.
	- Introduced a new `Settings` component to manage user settings, including routes for email confirmation.

- **Improvements**
	- Enhanced file upload processes across various components using the `uploadFile` function from the `useAtlasProvider`.
	- Improved email validation with a more robust regex pattern.
	- Added new utility functions for better toast notifications.
	- Streamlined the logic for managing visibility in navigation buttons.
	- Enhanced state management for the settings menu based on URL fragments.

- **Bug Fixes**
	- Streamlined error handling and notifications for file uploads and user settings changes.
	- Enhanced clarity in validation logic for various components.

- **Chores**
	- Removed several unused serverless functions and related TypeScript definitions to clean up the codebase.
	- Updated dependencies and environment variables for better configuration management.

### CODE REVIEW

File: web/src/context/AtlasProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      if (isValid && authToken) {
+        try {
+          const payload = decodeJwt(authToken);
+          const expiresIn = (payload.exp as number) * 1000 - Date.now();
+
+          timeoutId = setTimeout(verifyAndSchedule, Math.max(0, expiresIn));
+        } catch (err) {
+          console.error("Error decoding JWT:", err);
+          setIsVerified(false);
+        }
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/context/AtlasProvider.tsx:115]: try-catch catches it

File: web/src/context/AtlasProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+      const token = await loginUser(atlasGqlClient, { message, signature });
+      setAuthToken(token);
+    } catch (err: any) {
+      // eslint-disable-next-line
+      console.log("Authorize User Error : ", err?.message);
+    } finally {
+      setIsSigningIn(false);
+    }
+  }, [address, chainId, setAuthToken, signMessageAsync, atlasGqlClient]);
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/context/AtlasProvider.tsx:193]: The functions using it show those toast. We just invalidate the session with this wrapper

File: web/src/context/AtlasProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+      const payload = decodeJwt(authToken);
+
+      if ((payload?.sub as string)?.toLowerCase() !== address.toLowerCase()) return false;
+      if (payload.exp && payload.exp < Date.now() / 1000) return false;
+
+      return true;
+    } catch {
+      return false;
+    }
+  }, [authToken, address]);
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/context/AtlasProvider.tsx:96]: We don't need to do that. If it's tampered with then calls to backend will fail.

File: web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/index.tsx [hunk 4]
```diff
@@ -28,59 +36,66 @@ const ButtonContainer = styled.div`
...
+        email: emailInput,
+      };
+      addUser(data)
+        .then(async (res) => {
+          if (res) {
+            toggleIsSettingsOpen();
+          }
+        })
+        .catch((err) => console.log(err));
+    }
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/index.tsx:96]: Proper toasts are shown from within the addUser, updateUser  functions.

File: web/src/pages/Settings/index.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+const Settings: React.FC = () => {
+  return (
+    <Container>
+      <Routes>
+        <Route path="email-confirmation" element={<EmailConfirmation />} />
+      </Routes>
+    </Container>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/Settings/index.tsx:26]: The parent route handles that, if we needed we can handle here later. Not needed for now.

File: web/src/utils/atlas/confirmEmail.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  return client
+    .request<{ confirmEmail: ConfirmEmailResponse }>(query, variables)
+    .then(async (response) => response.confirmEmail)
+    .catch((errors) => {
+      // eslint-disable-next-line no-console
+      console.log("Confirm Email error:", { errors });
+
+      throw new Error("Unable to verify.");
+    });
+}
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/utils/atlas/confirmEmail.ts:36]: Check for address is done in the component. Additional checks are done in the call itself.

File: web/src/pages/SubmitItem/ItemField/FieldInput/ImageInput.tsx [hunk 7]
```diff
@@ -22,17 +21,12 @@ const StyledFileUploader = styled(FileUploader)`
...
-        const response = await res.json();
-        const imageURI = response["cids"][0];
-        handleWrite(imageURI);
+    uploadFile(file, Roles.CurateItemImage)
+      .then(async (fileURI) => {
+        if (!fileURI) return;
+        handleWrite(fileURI);
       })
       .catch((err) => console.log(err))
       .finally();
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/SubmitItem/ItemField/FieldInput/ImageInput.tsx:32]: file type and size validation will be added later, once those values are retrievable from atlas.

File: web/src/utils/atlas/updateEmail.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        // eslint-disable-next-line no-console
+        console.log("Update Email error:", { errors });
+
+        const error = errors?.response?.errors?.[0];
+
+        if (error) {
+          throw new GraphQLError(error?.message, { ...error });
+        }
+        throw new Error("Unknown Error");
+      }),
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/utils/atlas/updateEmail.ts:38]: Done in component utilizing this.

File: web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/EmailVerificationInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+      updateEmail({ newEmail: user.email })
+        .then(async (res) => {
+          if (res) {
+            toggleIsSettingsOpen();
+          }
+        })
+        .catch((err) => console.log(err));
+    },
+    [user, updateEmail, toggleIsSettingsOpen]
+  );
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/EmailVerificationInfo.tsx:80]: This button is used in a form, hence the necessary event prevention.

File: web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/index.tsx [hunk 10]
```diff
@@ -107,11 +122,18 @@ const FormContactDetails: React.FC<ISettings> = ({ toggleIsSettingsOpen }) => {
           isEditing={isEditingEmail}
         />
       </FormContactContainer>
-
+      {!isEmailUpdateable ? (
+        <StyledInfoCard msg={`You can update email again ${timeLeftUntil(user?.emailUpdateableAt!)}`} />
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/index.tsx:126]: isUndefined will catch null/ undefined values.

File: web/src/components/ActionButton/Modal/EvidenceUpload.tsx [hunk 11]
```diff
@@ -70,18 +70,16 @@ const EvidenceUpload: React.FC<IEvidenceUpload> = ({ setEvidence, setIsEvidenceU
...
+    uploadFile(file, Roles.Evidence)
+      .then(async (fileURI) => {
+        if (!fileURI) throw new Error("Error uploading file to IPFS");
         setFileURI(fileURI);
       })
-      .catch((err) => console.log(err))
+      .catch((err) => {
+        console.log(err);
+        toast.error(err?.message, toastOptions);
+      })
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/components/ActionButton/Modal/EvidenceUpload.tsx:82]: uploadFile handles that.

**Review by kemuru — APPROVED** [review]
everything seems to be working correctly. good job

File: web/src/pages/Settings/EmailConfirmation/index.tsx [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+  expired: {
+    headerMsg: "Verification link expired...",
+    subtitleMsg:
+      "Oops, the email verification link has expired. No worries! Go to settings and click on Resend it to receive another verification email.",
+    buttonMsg: "Open Settings",
+    buttonTo: "/#notifications",
+    Icon: WarningIcon,
+    color: "warning",
+  },
+};
```
Comment (alcercu) [inline web/src/pages/Settings/EmailConfirmation/index.tsx:128]: same as with the other PR, maybe this can be moved to its own file

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**